flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/failover/FailoverRegion.java

```diff
@@ -210,7 +212,7 @@ private void restart(long globalModVersionOfFailover) {
 					// we restart the checkpoint scheduler for
 					// i) enable new checkpoint could be triggered without waiting for last checkpoint expired.
 					// ii) ensure the EXACTLY_ONCE semantics if needed.
-					executionGraph.getCheckpointCoordinator().abortPendingCheckpoints(new Exception("FailoverRegion is restarting."));
+					executionGraph.getCheckpointCoordinator().abortPendingCheckpoints(new CheckpointException(CheckpointFailureReason.JOB_FAILURE));
```

Comment:
Please add some linebreak, this is a pretty long line.

Reply:
Also strictly speaking, this is not exactly a job failure, it is a partial failure of the job in one region. Thinking about if this might justify it's own failure cause.

Reply:
I have added a new failure reason named `JOB_FAILOVER_REGION` for this scene.